cohortextractor/validate_dummy_data.py

```diff
@@ -0,0 +1,131 @@
+from datetime import datetime
+
+import pandas as pd
+
+
+SUPPORTED_FILE_FORMATS = ["csv", "csv.gz"]
+
+
+class DummyDataValidationError(Exception):
+    pass
+
+
+def validate_dummy_data(column_definitions, dummy_data_file):
+    """Validate that dummy data provided by user matches expected structure and format.
+
+    Raises DummyDataValidationError if dummy data is not valid.
+    """
+
+    df = read_into_dataframe(dummy_data_file)
+    validate_expected_columns(df, column_definitions)
+
+    for col_name, query_node in column_definitions.items():
+        if col_name == "population":
+            # Ignore the population definition, since it is not used as a column in the
+            # output
+            continue
+
+        validator = get_csv_validator(col_name, query_node)
+
+        for ix, value in enumerate(df[col_name]):
+            if pd.isna(value) or not value:
```

Comment:
I think this will allow `0` as a value for something that should be a date (for example).

Reply:
It doesn't (but I had to write a test to check).  At this point, the `0` in the csv has been read in as a string, so `not "0"` is False

Reply:
Ah! Good to have the test anyway. Thanks.